src/Core/Scheduler/Providers/TileProvider.js

```diff
@@ -95,11 +94,11 @@ TileProvider.prototype.executeCommand = function executeCommand(command) {
     tile.layer = layer.id;
     tile.layers.set(command.threejsLayer);
 
-    if (parent) {
-        position.applyMatrix4(layer.object3d.matrixWorld);
-        parent.worldToLocal(position);
-        worldQuaternion.setFromRotationMatrix(parent.matrixWorld).inverse().multiply(layer.object3d.quaternion);
-        quaternion.premultiply(worldQuaternion);
+    tile.extentTransform = { quaternionInv: quaternion.clone().inverse(), position: position.clone() };
```

Comment:
At the very least add a `_` prefix.
But storing 2 extra properties, only to be able to compute the transform of children is not necessary.

Here's a version that also fixes the issue:
```js
    if (parent) {
        const p1 = new THREE.Vector3();
        const p2 = p1.clone();

        position.applyMatrix4(layer.object3d.matrixWorld);

        layer.object3d.matrixWorld.decompose(p1, worldQuaternion, p2);
        quaternion.premultiply(worldQuaternion);

        // Now position / quaternion are in World coordinates
        // But we want them in parent coordinates
        const m = new THREE.Matrix4();
        m.getInverse(parent.matrixWorld);
        parent.matrixWorld.decompose(p1, worldQuaternion, p2);

        // First, transform position
        position.applyMatrix4(m);
        // Second, transform quaternion
        worldQuaternion.inverse();
        quaternion.premultiply(worldQuaternion);
    }
```

(we have to use `decompose` and not `setFromRotationMatrix` because the latter is broken when scale is not 1)

Reply:
I also wanted to stop the use of `layer.object3d` which does not make much sense here

We can do this

```js
if (parent && parent instanceof TileMesh) {
        // get parent extent transformation
        const pTrans = builder.computeSharableExtent(parent.extent);
        // place relative to his parent
        position.sub(pTrans.position).applyQuaternion(pTrans.quaternion.inverse());
        quaternion.premultiply(pTrans.quaternion);
}
```

Reply:
> layer.object3d which does not make much sense here`

Using `layer.object3d` is mandatory if you use `parent.matrixWorld`, because `parent` is a descendant of `layer.object3d`.

But your alternative solution is looking good too.

Reply:
later I will calculate the position and quaternion relative to the parent in `computeSharableExtent`